hs_core/tasks.py

```diff
@@ -158,6 +168,108 @@ def nightly_periodic_task_check():
         file.write(timezone.now().strftime('%m/%d/%y %H:%M:%S'))
 
 
+@celery_app.task(ignore_result=True, base=HydroshareTask)
+def nightly_repair_resource_files():
+    """
+    Run repair_resource on resources updated in the last day
+    """
+    from hs_core.management.utils import check_time, repair_resource
+    start_time = time.time()
+    cuttoff_time = timezone.now() - timedelta(days=1)
+    recently_updated_resources = BaseResource.objects \
+        .filter(updated__gte=cuttoff_time, raccess__published=False)
+    repaired_resources = []
+    try:
+        for res in recently_updated_resources:
+            check_time(start_time, settings.NIGHTLY_RESOURCE_REPAIR_DURATION)
+            is_corrupt = False
+            try:
+                _, missing_django, dangling_in_django = repair_resource(res, logger)
+                is_corrupt = missing_django > 0 or dangling_in_django > 0
+            except ObjectDoesNotExist:
+                logger.info("nightly_repair_resource_files encountered dangling iRods files for a nonexistent resource")
+            if is_corrupt:
+                repaired_resources.append(res)
+
+        # spend any remaining time fixing resources that haven't been checked
+        # followed by those previously checked, prioritizing the oldest checked date
+        recently_updated_rids = [res.short_id for res in recently_updated_resources]
+        not_recently_updated = BaseResource.objects \
+            .exclude(short_id__in=recently_updated_rids, raccess__published=True) \
+            .order_by(F('files_checked').asc(nulls_first=True))
+        for res in not_recently_updated:
+            check_time(start_time, settings.NIGHTLY_RESOURCE_REPAIR_DURATION)
+            is_corrupt = False
+            try:
+                _, missing_django, dangling_in_django = repair_resource(res, logger)
+                is_corrupt = missing_django > 0 or dangling_in_django > 0
+            except ObjectDoesNotExist:
+                logger.info("nightly_repair_resource_files encountered dangling iRods files for a nonexistent resource")
+            if is_corrupt:
+                repaired_resources.append(res)
+    except TimeoutError:
+        logger.info(f"nightly_repair_resource_files terminated after \
+                    {settings.NIGHTLY_RESOURCE_REPAIR_DURATION} seconds")
+
+    if settings.NOTIFY_OWNERS_AFTER_RESOURCE_REPAIR:
+        for res in repaired_resources:
+            notify_owners_of_resource_repair(res)
+
+
+@shared_task
+def repair_resource_before_publication(res):
+    """
+    Run repair_resource on resource
+    """
+    from hs_core.management.utils import repair_resource
+    errors, missing_django, dangling_in_django = repair_resource(res, logger)
```

Comment:
Do we want to check here that the resource is not a published resource before calling `repair_resource`? This function is probably being already called for an unpublished resource.

Reply:
added a check to ensure resource is not already published